### PR TITLE
Remove date field

### LINKED ISSUE #1542: Edit view UI updates
- [x] Move description to the top on the toolbar like we have it in the start view.
- [x] When a user is editing the description, the action changes from "Save" to "Done".
- Selecting "Done" will bring them back into "Edit everything" mode where description field is not active (user is not currently editing the description).
- Selecting "Save" in "Edit everything" mode saves the changes and closes the edit view.
- [x] Tapping 'X' will discard the changes and closes the edit view. 'X' behaves the same way when the description field is not active and when it's active (when a user is currently typing the description). *Check confirmation #1543*
- [x] Remove the delete icon from the toolbar and move it at the bottom of the edit view list.
- [x] Tapping "Delete this entry" opens confirmation #1269 and tapping "Delete" once more, deletes this entry.
- [x] Show "Duration" label in the duration field.

### Edit view
edit

Zeplin: https://zpl.io/bPrLJQD

### PR DESCRIPTION
**Pull request by kornelijepetak**
Fixes the following points from #1542:

- [x] Remove date field
- [x] Show shortened date in the start/end time fields and use profile settings format. *For example: When user has set in the settings MM/DD/YYYY, then show only MM/DD*

Closes #1542 (since those are the last points)

### CODE REVIEW

**Comment by kornelijepetak** [comment]
I have added a WIP label because the revamping of the start/stop/duration has started, so if this is merged, there is no way to edit some of the fields.

**Comment by willsb** [comment]
One hint: I'd use a `Format` Value combiner in the bindings, but I'll let you decide if you wanna change that or not

**Comment by kornelijepetak** [comment]
I have put this PR `on-hold`.

⚠️Date field must not be removed until #1764 is implemented, otherwise there will be no way to set the date of a TE.

**Comment by willsb** [comment]
To the owner: Please take care not to reintroduce `Resource.designer.cs` when merging this.

**Comment by willsb** [comment]
#1764 is very much implemented, can we move on with this now? :)

**Comment by kornelijepetak** [comment]
@willsb This is no longer stale. I have merged develop and modified it as needed. Please rereview and we're done.

File: Toggl.Foundation.MvvmCross/ViewModels/EditTimeEntryViewModel.cs [hunk 1]
```diff
@@ -119,6 +119,8 @@ public TimeSpan DisplayedDuration
         [DependsOn(nameof(StopTime))]
         public bool IsTimeEntryRunning => !StopTime.HasValue;
 
+        public DateTimeOffset StopTimeOrCurrent => StopTime ?? default(DateTimeOffset);
```
Comment (willsb) [inline Toggl.Foundation.MvvmCross/ViewModels/EditTimeEntryViewModel.cs]: This is weird, why not use the time service, given the name of this property?

Comment (kornelijepetak) [reply Toggl.Foundation.MvvmCross/ViewModels/EditTimeEntryViewModel.cs]: oh, I just forgot to change this. I was trying something out and somehow it slipped :/

**Pull request merged**